### DISCUSSION TITLE
Nextflow cannot find Java or it's a wrong version but it seems like I have a higher version

### DISCUSSION THREAD

**Discussion by das2000sidd** [nextflow-io/nextflow#3225] [General]
Hi,

I am trying to get nextflow installed but I am running into a Java version error and it seems like I have a higher version of Java installed. Was wondering if somebody can help me resolve this bug.
Here is the output to java version check using java --version :

java 18.0.2.1 2022-08-18
Java(TM) SE Runtime Environment (build 18.0.2.1+1-1)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.2.1+1-1, mixed mode, sharing)

Here is the output to checking if nextflow works using ./nextflow run tutorial.nf

Error: cannot find Java or it's a wrong version -- please make sure that Java 7 or higher it's installed
Note: Nextflow is trying to use the Java VM defined by the following environment variables:
JAVA_CMD: /Library/Java/JavaVirtualMachines/jdk-18.0.2.1.jdk/Contents/Home/bin/java
JAVA_HOME:

I am using a OS X operating system.

**Comment by mribeirodantas** [comment]
I recommend using [sdkman ](https://sdkman.io) to install/manage Java. Could you please try and see if the paths are then correctly set so that nextflow can find Java?
